crates/mun_hir/src/ty/infer.rs

```diff
@@ -320,6 +321,12 @@ impl<'a> InferenceResultBuilder<'a> {
                                 lhs: *lhs,
                             })
                         }
+                        if !self.check_mut_bind(&resolver, *lhs) {
```

Comment:
I think it makes sense to turn this into an `else if` so that when you have something like `3=5` you don't get two errors.